csharp/src/SeedLang.Shell/Repl.cs

```diff
@@ -119,7 +119,8 @@ private class Visualizer : IVisualizer<AssignmentEvent>,
         WriteSource(visualizer.Source, syntaxTokens);
         Console.WriteLine("---------- Run ----------");
         var runCollection = new DiagnosticCollection();
-        executor.Run(visualizer.Source, "", _language, _runType, runCollection);
+        string source = visualizer.Source + Environment.NewLine;
```

Comment:
Just to confirm: does this change require every source code that is to be executed by SeedLang has to end with "\n"? For the case when SeedLang.Shell supports a feature to run a SeedPython file directly:

```
SeedLang.Shell.exe ... -c input.py
```
And user forgets to add the trailing "\n" to the input.py file. Will SeedLang.Shell adds a trailing NewLine to the input file automatically? Or, has the current grammar definition already covered the <EOF> issue when parsing simple statements?

Reply:
According to the definition of Python3 grammar:
```
file: [statements] ENDMARKER 
interactive: statement_newline
statements: statement+ 
statement: compound_stmt  | simple_stmts 
statement_newline:
    | compound_stmt NEWLINE 
    | simple_stmts
    | NEWLINE 
    | ENDMARKER 
simple_stmts:
    | simple_stmt !';' NEWLINE  # Not needed, there for speedup
    | ';'.simple_stmt+ [';'] NEWLINE
```
Python uses `file` and `interactive` rules to distinguish these two different use cases. I think it might not be needed now, one `program` rule is enough to cover these two. I need verify it when the file running case is implemented.
And from above grammar definition, it seems `NEWLINE` is needed for `simple_stmt`, but not needed for `compound_stmt`. I need to verify it as well.
Let's keep it open, and finalize it when the file running case is implemented. Anyway the solution could be adding NEWLINE at the end of the file if users miss it.

Reply:
If it's the case "anyway the solution could be adding NEWLINE at the end of the file if users miss it", why don't we check and add the missing NEWLINE inside the logic of executor.Run() for SeedPython?

Reply:
There is one problem to add NEWLINE inside SeedLang library, because ANTLR will report an error range if there is errors happening, and sometimes the NEWLINE is included in the error range. SeedLang.Shell can not handle the error range correctly in this case if the source is modified by SeedLang library.

Reply:
OK. I see. Let's move forward and try to solve this in a better way later.